### ISSUE TITLE
amp-list - layout "container"

### ISSUE THREAD

**Issue by antoinebisch** [ampproject/amphtml#21517]
## Describe the new feature or change to an existing feature you'd like to see

I am currently implementing a dynamic cart in AMP and the page only contains a header, an amp-list element and a footer.
The issue is that the amp-list content is generated at load-time and because it can contain X products, needs to be dynamically sized. However amp-list does not support "container" layout, so only "fill" can be used here. This positions the amp-list as absolute which collapses its content with the footer.

## Describe alternatives you've considered

I have tried overriding the absolute position of amp-list and its generated elements but this results in the amp-list not generating the elements in the template.
Current alternative is to fix the position of the footer at the bottom of the page and super-impose it on top of the amp-list element with a higher z-index value. However the footer is high and hides most of the amp-list content so this is not a great solution.

## Additional context

A demo can de found [here](https://amp-list-collapse.glitch.me/). Please add the item to the cart and observe the issue on the cart page redirect.

This is also related to this previously submitted [FR](https://github.com/ampproject/amphtml/issues/7000).

**Comment by cathyxz** [comment]
We explored the need for dynamic resizing in amp-list and ended up implementing the `changeToLayoutContainer` action. 

In your particular use case, what people have generally done is use `amp-list` with layout `fixed-height`, where you write your page so that the height of the list is either default below the viewport or within the bottom 15% / 1000px of the page end--in that case, amp-list will dynamically resize based on the height of its contents. Using `layout=fixed-height`, if the above conditions are not satisfied, amp-list will try but fail to resize and end up showing an overflow button. 

Details regarding the overflow condition was discussed in https://github.com/ampproject/amphtml/issues/14060#issuecomment-466099859, but I agree that our current API is non-ideal for this use case, and this is definitely something we're looking into / trying to solve for a future version of amp-list. 

/cc @nainar

**Comment by morsssss** [comment]
Thanks! This is one of the biggest issues we've faced in our e-comm demo... a standard site feature that we simply can't create with standard AMP components.

**Comment by mattwomple** [comment]
Thanks for the ping @morsssss . [#10230 amp-list: Support dynamic resizing](https://github.com/ampproject/amphtml/issues/10230) has a lot of relevant discussion that would be good to consider when fleshing-out the design.

We certainly would appreciate this feature at WompMobile, as many of our clients are eCommerce and product listings are rarely of constant height. We workaround the problem with a little CSS right now, so it technically is possible to have "variable height" amp-lists, but it's not the most elegant solution. Official support would be welcome!

**Comment by morsssss** [comment]
Thanks for the link, @mattwomple ! Madison and I had occasion to talk about your CSS solution again last week, and if we do make this public it will at least provide others with a way forward! But of course it would be really nice if this could be supported out of the box.

**Comment by palewire** [comment]
This would be a great feature to add. If I understand the situation correctly, there is currently way no documented way to reliably position any elements at all below a long, unfurled Google `amp-list`?

**Comment by morsssss** [comment]
@nainar I just ran across this issue again, and I thought I'd ping it up. I'm not sure if any `<amp-list>` enhancements are currently underway... can you advise?

**Comment by cathyxz** [comment]
I think there's a lot of issues for this feature request already. @caroqliu could I trouble you to dedupe this against whichever most recent issue you're using to track this feature? =)

**Comment by caroqliu** [comment]
Thanks @cathyxz 😅  Dedup with #26873
